Simplenote/SharedStorageMigrator.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import CoreData
+
+@objc
+class SharedStorageMigrator: NSObject {
+    let storageSettings: StorageSettings
+
+    @objc
+    init(storageSettings: StorageSettings) {
+        self.storageSettings = storageSettings
+    }
+
+    /// Database Migration
+    /// To be able to share data with app extensions, the CoreData database needs to be migrated to an app group
+    /// Must run before Simperium is setup
+
+    @objc
+    func performMigrationIfNeeded() {
+        // Confirm if the app group DB exists
+        guard migrationNeeded else {
+            NSLog("Core Data Migration not required")
+            return
+        }
+
+        migrateCoreDataToAppGroup()
+    }
+
+    var migrationNeeded: Bool {
+        return storageSettings.legacyStorageExists && !storageSettings.sharedStorageExists
+    }
+
+    func migrateCoreDataToAppGroup() {
+        // Testing prints
+        // TODO: Remove prints later
+        print("oldDb exists \(FileManager.default.fileExists(atPath: storageSettings.legacyStorageURL.path))")
+        print(storageSettings.legacyStorageURL.path)
+        print("newDb exists \(FileManager.default.fileExists(atPath: storageSettings.sharedStorageURL.path))")
+        print(storageSettings.sharedStorageURL.path)
+
+        NSLog("Database needs migration to app group")
+        NSLog("Beginning database migration")
+
+        do {
+            try disableJournaling()
+            try migrateCoreDataFiles()
+            NSLog("Database migration successful!!")
+        } catch {
+            NSLog("Could not migrate database to app group")
+            NSLog(error.localizedDescription)
+        }
+    }
+
+    private func disableJournaling() throws {
+        guard let mom = NSManagedObjectModel(contentsOf: storageSettings.modelURL) else {
+            return
+        }
+        let psc = NSPersistentStoreCoordinator(managedObjectModel: mom)
+
+        let options = [NSSQLitePragmasOption: ["journal_mode": "DELETE"]] as [AnyHashable: Any]
+
+        try psc.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: storageSettings.legacyStorageURL, options: options)
+    }
+
+    private func migrateCoreDataFiles() throws {
+        try FileManager.default.copyItem(at: storageSettings.legacyStorageURL, to: storageSettings.sharedStorageURL)
```

Comment:
Why not `moveItem`? Do we plan to do something with the old database file? 🤔

Reply:
@jleandroperez and I discussed it and thought it might be good to have the file remain in place in case something bad happens, we can restore the original DB url without needing to migrate back.  The old DB might be a bit out of date, but it would still be there.

Reply:
We may want to use `NSPersistentStoreCoordinator`. Last 2020, we had a chat with one of the Apple devs. Quoting what Jaclyn summarized (p91TBi-2YF-p2): 

> They did not recommend backing up the existing database like we do now, copying and removing the SQLite file upon the first persistent store loading error. In order to actually back up a persistent store, it takes three files: *.sqlite, *.sqlite-shm, and *.sqlite-wal. Mutating the database files with file system manger is also not safe, since there could be other processes accessing the persistent store like from extensions. They recommended backing up the persistent store via NSPersistentStoreCoordinator‘s replacePersistentStore and removing it via destroyPersistentStore.

I'm not familiar with disabling journaling. Does disabling it remove the WAL files? 

But I guess even then, it's probably best to try and use one of the available APIs that Apple provided for us. We've had quite a lot of crashes in Woo in the past and changing our `FileManager` usages to `NSPersistentStoreCoordinator` functions have drastically reduced those crashes.  

I haven't used this before but it looks like [`migratePersistentStore`](https://developer.apple.com/documentation/coredata/nspersistentstorecoordinator/3747534-migratepersistentstore) may be an option instead of `FileManager`. If not, maybe [`NSMigrationManager.migrate`](https://developer.apple.com/documentation/coredata/nsmigrationmanager/3747519-migratestore) with the same source and destination models. 😅

Reply:
@shiki BTW (sorry I'm late), **thank you** for checking this PR sir!!